galaxy_importer/constants.py

```diff
@@ -58,7 +58,13 @@
 # Matches role names with any combination of lowercase letters,
 # uppercase letters, numbers, underscores, and hyphens with
 # a length in the inclusive range [1, 55].
-LEGACY_ROLE_NAME_REGEXP = re.compile("^[a-zA-Z0-9-_]{1,55}$")
+LEGACY_ROLE_NAME_REGEXP = re.compile("^[a-zA-Z0-9_-]{1,55}$")
+# Matches namespaces: Namespace names should match any valid
+# GitHub username. Username may only contain alphanumeric characters
+# or single hyphens, and cannot begin or end with a hyphen
+# Some roles may contain underscores
+# Maximum of 39 char tested in the validator
+LEGACY_NAMESPACE_REGEXP = re.compile("^([a-zA-Z0-9]+[-_]?)*[a-zA-Z0-9]$")
```

Comment:
This neesd to include namespace names that end with `-` and contain `.`

Reply:
@chouetz I think this has become a critical issues. I suspect you have a busy schedule and can't devote your attention to this issue, so I'm going to give it till the end of the day for you to address my review comments. If you are unable to, I'll try to push fixes to your branch or create a new PR with your commits cherry picked. Thanks for the help on this one, it's greatly appreciated.

Reply:
I've just updated the PR right now, here is the output of your gist with the updated regex
```
(venv) {15:54:19}nicolas.schweitzer@mac~/Downloads/galaxy ➭ python check_roles_regex.py
FAILED - id:6949 namespace:_Harsha_
FAILED - id:11063 namespace:kasunmaduraeng@gmail.com
FAILED - id:14822 namespace:mhabrnal@redhat.com
FAILED - id:14826 namespace:–role-name
FAILED - id:15553 namespace:/etc/ansible/roles/tt-test.nfs
FAILED - id:16048 namespace:rohitggarg/docker-swarm#
FAILED - id:16050 namespace:docker-swarm/tree/master/roles
FAILED - id:18892 namespace:childbdm@gmail.com
FAILED - id:21867 namespace:shashikanth.komandoor@gmail.com
FAILED - id:21913 namespace:ggriffin924@gmail.com
FAILED - id:22855 namespace:piotrfaj@gmail.com
FAILED - id:23153 namespace:farandir@gmail.com
FAILED - id:23178 namespace:lifeblood@163.com
FAILED - id:24233 namespace:ajayshankeryadav@yahoo.co.in
FAILED - id:25461 namespace:ch680351034@gmail.com
FAILED - id:26481 namespace:mheiges@gmail.com
```
I've just allowed names ending with underscores on top of your requirement

Reply:
ty!